pkg/auth/middleware_test.go

```diff
@@ -0,0 +1,126 @@
+package auth_test
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/guidewire/fern-reporter/pkg/auth"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"net/http"
+	"net/http/httptest"
+)
+
+var _ = Describe("JWK Management", func() {
+	var (
+		server *ghttp.Server
+		url    string
+	)
+
+	BeforeEach(func() {
+		// Create a fake server to simulate JWK provider
+		server = ghttp.NewServer()
+		url = server.URL()
+	})
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	Describe("UpdateJWKS", func() {
+		Context("when fetching JWKs is successful", func() {
+			BeforeEach(func() {
+				server.AppendHandlers(
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/"),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, map[string]interface{}{
```

Comment:
It would be cleaner to use the JWK struct here, instead of `map[string]interface{}` inline.